src/os/ui/config/abstractsettings.js

```diff
@@ -52,6 +52,16 @@ os.ui.config.AbstractSettingsCtrl = function($scope, $timeout) {
    */
   this.scope['selected'] = null;
 
+  /**
+   * @type {boolean}
+   */
+  this.scope['isResetDisabled'] = false;
+
+  /**
+   * @type {string}
+   */
+  this.scope['resetDisabledText'] = 'You are currently not allowed to Reset Settings.';
```

Comment:
Is it possible to provide more information here? This should already be obvious from the disabled button.

Reply:
I kept this generic so that whomever extends it can set the disabled text appropriately based on why they disabled the button.  For example, in bits-internal I set it to more specific text: 'In order to Reset Settings, you may not have the same application opened multiple times.  Please close all but one, and retry.'  I can't think of how to provide more detailed text in this file.

Reply:
You could disable the button when `resetDisabledText` is truthy and set the default value to an empty string. Then you wouldn't need `isResetDisabled` at all, and the override could set `resetDisabledText` to an appropriate message when needed.

Reply:
Great idea.  Thanks!